aws-auth-cognito/src/main/java/com/amplifyframework/statemachine/StateMachine.kt

```diff
@@ -24,9 +25,15 @@ import kotlinx.coroutines.Job
 import kotlinx.coroutines.launch
 import kotlinx.coroutines.newFixedThreadPoolContext
 
-internal typealias StateChangeListenerToken = UUID
 internal typealias OnSubscribedCallback = () -> Unit
 
+@JvmInline
+internal value class StateChangeListenerToken private constructor(val uuid: UUID) {
+    companion object {
+        fun create() = StateChangeListenerToken(UUID.randomUUID())
```

Comment:
Since this is an internal class, should we just create via `StateChangeListenerToken()` with a default uuid value and remove the factory?

Reply:
+1